tests/integration/helpers.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from typing import Optional
+
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+
+async def get_address(ops_test: OpsTest, app_name: str, unit_num: Optional[int] = None) -> str:
+    """Find unit address for any application.
+
+    Args:
+        ops_test: pytest-operator plugin
+        app_name: string name of application
+        unit_num: integer number of a juju unit
+
+    Returns:
+        unit address as a string
+    """
+    status = await ops_test.model.get_status()
+    app = status["applications"][app_name]
+    return (
+        app.public_address
+        if unit_num is None
+        else app["units"][f"{app_name}/{unit_num}"]["address"]
+    )
```

Comment:
@addyess Fulfilling my promise to teach you when I found out, this is how you obtain the external IP.

Reply:
so the application status has the address?  good to know